src/curve/constant-product/ConstantProductPair.sol

```diff
@@ -167,8 +148,10 @@ contract ConstantProductPair is ReservoirPair {
         amount1 = liquidity * balance1 / _totalSupply; // using balances ensures pro-rata distribution
         require(amount0 > 0 && amount1 > 0, "CP: INSUFFICIENT_LIQ_BURNED");
         _burn(address(this), liquidity);
-        _safeTransfer(_token0, to, amount0);
-        _safeTransfer(_token1, to, amount1);
+
+        _returnAndTransfer(_token0, to, amount0, _reserve0, _reserve1);
+        _returnAndTransfer(_token1, to, amount1, _reserve0, _reserve1);
```

Comment:
Sounds like function transfer, this naming confused me a lot. Suggest changing it to something else. Could do `_checkedTransfer`

Reply:
okay fixed